pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -821,3 +846,40 @@ func MakeNextPodFunc(queue SchedulingQueue) func() *framework.QueuedPodInfo {
 func podInfoKeyFunc(obj interface{}) (string, error) {
 	return cache.MetaNamespaceKeyFunc(obj.(*framework.QueuedPodInfo).Pod)
 }
+
+// Checks if the Pod may become schedulable upon the event.
+// This is achieved by looking up the global clusterEventMap registry.
+func (p *PriorityQueue) podMatchesEvent(podInfo *framework.QueuedPodInfo, event string) bool {
+	clusterEvent, ok := clusterEventReg[event]
+	if !ok {
+		return false
+	}
+	if clusterEvent == framework.WildCardEvent {
+		return true
+	}
+
+	evtToPlugins := p.clusterEventMap[podInfo.Pod.Spec.SchedulerName]
+	if evtToPlugins == nil {
+		return false
+	}
+
+	// Don't directly use evtToPlugins[clusterEvent].Intersection(podInfo.UnschedulablePlugins) as
+	// upon the same resource type, we need to check if the ActionType is *compatible* instead
+	// of *identical*.
+	for evt, nameSet := range evtToPlugins {
+		if evt.Resource != clusterEvent.Resource {
+			continue
+		}
+
+		// Update AND sub-type of Update gets a non-zero number, so does
+		// any type (rather than Unknown) AND All. That implies it's a match.
+		if evt.ActionType&clusterEvent.ActionType != 0 {
+			// Do NOT return directly.
+			if len(nameSet.Intersection(podInfo.UnschedulablePlugins)) != 0 {
```

Comment:
`Intersection` instantiates a map, can we avoid that?

Also, it looks like we don't need `nameSet` to be a `sets.String`. It could just be a `[]string` to save lookups. But perhaps a performance would help us inform this decision.

Reply:
Yes. The Intersection library here is less performant, will implement our own "Intersection" if we stick with map.
I will resolve other comments first, and run perf test to come up with the final decision - slice or map.